src/main/scala/com/fulcrumgenomics/umi/FilterConsensusReads.scala

```diff
@@ -294,6 +307,15 @@ class FilterConsensusReads
       val depths2 = rec[Array[Short]](BaRawReadCount)
       val errors1 = rec[Array[Short]](AbRawReadErrors)
       val errors2 = rec[Array[Short]](BaRawReadErrors)
+      val bases1  = rec.get[String](AbConsensusBases)
+      val bases2  = rec.get[String](BaConsensusBases)
+      val singleStrandDisagreements: Seq[Boolean] = { // pre-compute where the two single-strand consensus sequences disagree
```

Comment:
I had an idea for this!  How about a local function:

```scala
def ssAgrees: Int => Boolean = (bases1, bases2) match {
  case (Some(s1), Some(s2)) => (idx) => s1(idx) == s2(idx)
  case _                    => (idx) => true
}
```

Then below you can write:

```scala
val singleStrandDisagree = this.requireSingleStrandAgreement && !ssAgree(i)
```